src/hibernate-resume/hibernate-resume-generator.c

```diff
@@ -29,6 +35,17 @@ STATIC_DESTRUCTOR_REGISTER(arg_resume_device, freep);
 STATIC_DESTRUCTOR_REGISTER(arg_resume_options, freep);
 STATIC_DESTRUCTOR_REGISTER(arg_root_options, freep);
 
+#if ENABLE_EFI
+typedef struct EFIHibernateLocation {
+        sd_id128_t uuid;
+        uint64_t offset;
+        char *id;
+        char *image_id;
+        char *version_id;
+        char *image_version;
```

Comment:
these four fields should be `const char*`, given you aprse them with json_dispatch_const_string